apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/utils.ts

```diff
@@ -0,0 +1,975 @@
+  const [highPriorityFee, mediumPriorityFee, lowPriorityFee] =
+    await Promise.all([
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.HIGH),
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.MEDIUM),
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.LOW),
+    ]);
+
+  // Select target fee rate based on user priority
+  const targetFeeRate = selectTargetFeeRate(feePriority, {
+    high: highPriorityFee,
+    medium: mediumPriorityFee,
+    low: lowPriorityFee,
+  });
+
+  // Validate inputs
+  validateTransactionInputs(txHex, fee, availableUtxos);
+
+  // Create analyzer with wallet-specific parameters
+  const analyzer = new TransactionAnalyzer({
+    txHex,
+    network,
+    targetFeeRate,
+    absoluteFee: fee.toString(),
+    availableUtxos,
+    requiredSigners: walletConfig.requiredSigners,
+    totalSigners: walletConfig.totalSigners,
+    addressType: walletConfig.addressType,
+  });
+
+  // Get comprehensive analysis
+  const analysis = analyzer.analyze();
+
+  // Return the analysis from the analyzer with added network fee estimates
+  return {
+    ...analysis,
+    networkFeeEstimates: {
+      highPriority: highPriorityFee,
+      mediumPriority: mediumPriorityFee,
+      lowPriority: lowPriorityFee,
+    },
+    userSelectedFeeRate: targetFeeRate, // Telling us the feeRate we can expect based on user selected no. of blocks to confirm
+    userSelectedPriority: feePriority, // Telling us which FeePriority user choosed
+  };
+};
+
+// ============================================================================
+// UTXO EXTRACTION AND FORMATTING
+// ============================================================================
+
+/**
+ * Extracts UTXOs from a transaction and wallet state for fee bumping
+ *
+ * This function collects two types of UTXOs needed for fee bumping operations:
+ * 1. Original transaction inputs (required for RBF) - with recovered BIP32 paths and scripts
+ * 2. Available wallet UTXOs (for adding inputs to new transactions)
+ *
+ * For spent UTXOs, this function implements a recovery mechanism to restore
+ * BIP32 derivation paths and multisig scripts by matching output addresses
+ * to known wallet slices.
+ *
+ * @param transaction - The transaction object to analyze
+ * @param walletState - Current wallet state containing UTXOs
+ * @param blockchainClient - Blockchain client for fetching additional details if needed
+ * @returns Array of UTXOs formatted for use with the fee bumping algorithms
+ */
+export const extractUtxosForFeeBumping = async (
+  transaction: TransactionT,
+  depositNodes: any,
+  changeNodes: any,
+  blockchainClient: BlockchainClient,
+): Promise<FeeUTXO[]> => {
+  // Array to store
+  const utxos: FeeUTXO[] = [];
+  try {
+    // STEP 0: PREPARE WALLET SLICES FOR BIP32 RECOVERY
+    // ------------------------------------------------
+    // Combine all wallet slices for efficient lookup during BIP32 recovery
+    const allSlices = [
```

Comment:
why not just have one arg for the function for allSlices and do this when calling the function? This function is already very complex so any opportunities to simplify would be worthwhile.